### ISSUE TITLE
rules fail to apply on intersection types

### ISSUE THREAD

**Issue by Znerole** [cartant/eslint-plugin-rxjs#60]
When intersecting `Observable` with other types (https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types), the rules fail to apply, apparently they don't recognize the type. I encountered this with "no-ignored-error", but I suppose other rules are affected, too.

Example:

```typescript
import { Observable } from "rxjs"

interface ISomeExtension {
   sayHi(): void
}

function subscribeObservable<T>(obs: Observable<T>) {
   return obs.subscribe((v: T) => {}) // eslint error: rxjs/no-ignored-error
}

function subscribeObservableWithExtensio<T>(obs: Observable<T> & ISomeExtension) {
   return obs.subscribe((v: T) => {}) // no eslint error
}
```

Function `subscribeObservable` fails as expected, `subscribeObservableWithExtension` passes eslint successfully (which it shouldn't).

Package versions:
* eslint-plugin-rxjs@2.1.5
* eslint@7.17.0
* typescript@4.1.3

**Comment by cartant** [comment]
The underlying implementation of `couldBeType` does check for intersections, but the rule itself was unnecessarily insisting on type references, so this would have effected a failure:

```ts
import { Observable } from "rxjs"

interface ISomeExtension {
   sayHi(): void
}

type SomeExtendedObservable<T> = Observable<T> & ISomeExtension;

function subscribeObservableWithExtensio<T>(obs: SomeExtendedObservable<T>) {
   return obs.subscribe((v: T) => {})
}
```

I've removed the type reference checks, so this should be fixed.

**Comment by Znerole** [comment]
Awesome, thank you!
